exporter/collector/metricsexporter.go

```diff
@@ -162,23 +162,33 @@ func (me *metricsExporter) pushMetrics(ctx context.Context, m pdata.Metrics) err
 		for j := 0; j < ilms.Len(); j++ {
 			ilm := ilms.At(j)
 
-			extraLabels := mergeLabels(instrumentationLibraryToLabels(ilm.InstrumentationLibrary()), extraResourceLabels)
+			instrumentationLibraryLabels := me.mapper.instrumentationLibraryToLabels(ilm.InstrumentationLibrary())
+
 			mes := ilm.Metrics()
 			for k := 0; k < mes.Len(); k++ {
 				metric := mes.At(k)
+				metricLabels := extraResourceLabels
+
+				if me.mapper.shouldAddInstrumentationLibraryLabels(metric) {
```

Comment:
We made this same logic for resources, actually (@dashpole or @aabmass may have the most context on system-metrics being recorded with our APIs).